File: app/src/main/java/com/example/furnitures/calculator/furniture/FurnitureTypeHelper.java
package com.example.furnitures.calculator.furniture;

import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;
import com.example.furnitures.R;

/**
 * Copyright (c) 2017 fluidmobile GmbH. All rights reserved.
 */
public final class FurnitureTypeHelper {

    private FurnitureTypeHelper() {
    }

    @DrawableRes
    public static int getDrawable(@FurnitureType int furnitureType) {
        switch (furnitureType) {
            case FurnitureType.ARMCHAIR:
                return R.drawable.ic_furniture_armchair;
            case FurnitureType.BED:
                return R.drawable.ic_furniture_bed;
            case FurnitureType.SOFA:
                return R.drawable.ic_furniture_sofa;
            case FurnitureType.UNDEFINED:
            default:
                throw new IllegalStateException("\"Could not find Drawable for furniteType \" + furnitureType");
        }
    }

    @StringRes
    public static int getString(@FurnitureType int furnitureType) {
        switch (furnitureType) {
            case FurnitureType.ARMCHAIR:
                return R.string.furniture_armchair;
            case FurnitureType.BED:
                return R.string.furniture_bed;
            case FurnitureType.SOFA:
                return R.string.furniture_sofa;
            case FurnitureType.UNDEFINED:
            default:
                throw new IllegalStateException("Could not find String for furniteType " + furnitureType);
        }
    }
}

File: app/src/main/java/com/example/furnitures/calculator/trick/FurnitureTypeHelper.kt
package com.example.furnitures.calculator.trick

import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import com.example.furnitures.R

object FurnitureTypeHelper {

    // BUG: multiple pictures selects, when you click different items with the same drawable
    @DrawableRes
    fun getDrawable(furnitureType: FurnitureType): Int {
        when (furnitureType) {
            FurnitureType.ARMCHAIR -> return R.drawable.ic_furniture_armchair
            FurnitureType.BED -> return R.drawable.ic_furniture_bed
            FurnitureType.SOFA -> return R.drawable.ic_furniture_sofa
            FurnitureType.USER_CREATED_GRINDE -> return R.drawable.ic_letter_g
            FurnitureType.USER_CREATED_SLIDE -> return R.drawable.ic_letter_s
            FurnitureType.USER_CREATED_OTHER -> return R.drawable.ic_letter_o
            FurnitureType.UNDEFINED -> throw IllegalStateException("\"Could not find Drawable for furniteType \" + furnitureType")
            else -> throw IllegalStateException("\"Could not find Drawable for furniteType \" + $furnitureType")
        }
    }

    @StringRes
    fun getString(furnitureType: FurnitureType): Int? {
        when (furnitureType) {
            FurnitureType.ARMCHAIR -> return R.string.furniture_armchair
            FurnitureType.BED -> return R.string.furniture_bed
            FurnitureType.SOFA -> return R.string.furniture_sofa
            FurnitureType.USER_CREATED_GRINDE -> return null
            FurnitureType.USER_CREATED_SLIDE -> return null
            FurnitureType.USER_CREATED_OTHER -> return null
            FurnitureType.UNDEFINED -> throw IllegalStateException("Could not find String for furniteType $furnitureType")
            else -> throw IllegalStateException("Could not find String for furniteType $furnitureType")
        }
    }
}
